### PR TITLE
scylla-bench: add user mode

### PR DESCRIPTION
**Pull request by rjeczalik**
This PR adds initial support for user mode benchmarking - running performance test guided by a profile file.

New `usermode` package is introduced, which is responsible for parsing profile files (`usermode.Profile` model) and running the test (`usermode.Driver`).

There are still a number of rough edges, like missing support for all the distribution types, column data types - all of those are documented as inline godoc comments in respective places.

Please take a look.

/cc @dahankzter

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -187,7 +188,7 @@ func main() {
 
 	flag.StringVar(&nodes, "nodes", "127.0.0.1", "nodes")
 	flag.BoolVar(&clientCompression, "client-compression", true, "use compression for client-coordinator communication")
-	flag.IntVar(&concurrency, "concurrency", 16, "number of used goroutines")
+	flag.IntVar(&concurrency, "concurrency", runtime.GOMAXPROCS(0), "number of used goroutines")
```
Comment (dahankzter) [inline main.go]: Why not runtime.NumCPU()?

Comment (rjeczalik) [reply main.go]: Yeah, `runtime.NumCPU` is a better fit. Changed.

File: usermode/distribution.go [hunk 2]
```diff
@@ -71,14 +71,16 @@ func NewGenerator() *Generator {
...
-// The seed argument controls the uniqueness distribution of the generated
+// a unique seed was generated for the argument and it successfully filled with random data,
+// false otherwise.
+// If uniq argument is false, the function reuses the seeds and always returns true.
+// The col.Population argument controls the uniqueness distribution of the generated
 // data.
-// The size argument controls the size in bytes of the generated data.
-func (g *Generator) Generate(col *ColumnSpec, out interface{}) bool {
+// The col.Size argument controls the size in bytes of the generated data.
+func (g *Generator) Generate(col *ColumnSpec, out interface{}, uniq bool) bool {
```
Comment (dahankzter) [inline usermode/distribution.go]: Is this and the following commits related to the "duration" addition?

Comment (rjeczalik) [reply usermode/distribution.go]: Indirectly yes - testing the tool with duration (hundred of thousands iterations) shown an issue, when we run out of unique seeds for columns that we do not care whether data generated is unique (non-partition and non-cluster columns). This issue manifested itself with gocql errors like:

```
2018/09/18 08:18:28 gocql: batch statement 7 expected 6 values send got 5
2018/09/18 08:18:29 gocql: batch statement 3 expected 6 values send got 5
```

**Pull request closed without merging**